### PR TITLE
81881 cie vaos get clinic implementation

### PR DESCRIPTION
**Pull request by kanchanasuriya**
## Summary

- *Get Clinic method implmentation for CIE VAOS appointments endpoint*

## Related issue(s)

- *https://app.zenhub.com/workspaces/check-in-experience-61fc23a2cb8a14001132e102/issues/gh/department-of-veterans-affairs/va.gov-team/81881*

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
NA

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: modules/check_in/app/services/check_in/vaos/facility_service.rb [hunk 1]
```diff
@@ -39,6 +39,13 @@ def get_facility
         end
       end
 
+      def get_clinic(clinic_id:)
+        with_monitoring do
+          response = perform(:get, facilities_base_path + "/clinics/#{clinic_id}", {}, headers)
```
Comment (randomsync) [inline modules/check_in/app/services/check_in/vaos/facility_service.rb:44]: I believe the clinics are subbed under a facility, so this method will need to take a facility_id as a parameter too

Comment (kanchanasuriya) [reply modules/check_in/app/services/check_in/vaos/facility_service.rb:44]: Yeah, thats right, but facility_id is already available in the facility_service object.  So we do not need to pass the facility_id to this method.

Comment (randomsync) [reply modules/check_in/app/services/check_in/vaos/facility_service.rb:44]: I didn't realize that we're initializing the class with facility_id. Do we need to do that? I think the class can be generic and the facility_id and clinic_id can be passed as parameters. This way, the controller code would be simplified as well.

Comment (kanchanasuriya) [reply modules/check_in/app/services/check_in/vaos/facility_service.rb:44]: Agree, Gaurav.  I will update the PR to move the facility_id to parameter.

Comment (kanchanasuriya) [reply modules/check_in/app/services/check_in/vaos/facility_service.rb]: Pls find the updated code in commit [f662c10](https://github.com/department-of-veterans-affairs/vets-api/pull/16619/commits/f662c10531d2bbe8dbeb3abe1aed790313830782)

File: modules/check_in/app/services/check_in/vaos/facility_service.rb [hunk 2]
```diff
@@ -13,28 +13,16 @@ class FacilityService < Common::Client::Base
...
+          response = perform(:get, facilities_base_path(facility_id:), {}, headers)
+          response.body
+        end
       end
 
-      def get_facility
+      def get_clinic(facility_id:, clinic_id:)
         with_monitoring do
-          response = perform(:get, facilities_base_path, {}, headers)
+          response = perform(:get, facilities_base_path(facility_id:) + "/clinics/#{clinic_id}", {}, headers)
```
Comment (randomsync) [inline modules/check_in/app/services/check_in/vaos/facility_service.rb:25]: I'd suggest having a `clinics_url(facility_id:, clinic_id:)` method

File: modules/check_in/app/services/check_in/vaos/facility_service.rb [hunk 3]
```diff
@@ -45,7 +33,7 @@ def config
 
       private
 
-      def facilities_base_path
+      def facilities_base_path(facility_id:)
```
Comment (randomsync) [inline modules/check_in/app/services/check_in/vaos/facility_service.rb:36]: base_path might be misleading. would suggest calling is `facilities_url`

**Comment by randomsync** [comment]
are you planning to implement caching in this PR or another?

**Review by randomsync — APPROVED** [review]

**Pull request merged**